torch/jit.py

```diff
@@ -1,26 +1,375 @@
 import os
+import functools
 import torch.contrib._graph_vis as graph_vis
-# Example how to use:
-#
-# import torch.jit
-# model = model.RNNModel(args.model, ...)
-# model = torch.jit.traced(model)
 
 
-def flatten(x):
+HOLE = object()
+VOLATILE = object()
+
+
+# TODO: verify is not implemented yet
+def compile(arg=None, nderivs=1, params=tuple(), verify=False, optimize=True, enabled=True):
+    """
+    Mark a function or module as eligible for just-in-time compilation.  The
+    next time the function/module is executed, it is traced, and the trace is
+    compiled into an optimized representation which is run in lieu of the
+    original Python code upon subsequent invocations of the function/module.
+
+    The result of this function is stateful, so make sure you invoke it
+    only once per code you want to JIT compile.
+
+    .. note::
+
+        If your function/module takes non-Variable inputs, the JIT compiler
+        will compile a trace separately for each distinct input configuration.
+
+    .. warning::
+
+        Just-in-time compilation currently only works for functions/modules
+        which do not have dynamic control flow; if you compile a function/module
+        which has dynamic control flow, you will silently get incorrect
+        results on subsequent invocations of the model.  Use `verify=True` to
+        check that the original Python code and optimized code are equivalent.
+
+    Arguments:
+        arg (optional, torch.nn.Module or function): the function or module
+            to be compiled.  If `None`, `compile` returns a decorator which can be
+            applied to the function or module you want to compile.
+        verify (bool, default False): if True, upon all invocations of the
+            function/module, execute both the compiled and interpreted versions
+            of the model, and verify that their results match.  This is an easy
+            (albeit slow) way to check if your function/module can be validly
+            JIT compiled.
+        nderivs (int, default 1): the number of derivatives which this function/module
+            will be used with.  You MUST accurately specify this number: set it too
+            low and you will see an error when you attempt to run `backward`;
+            set it too high, and the function/module will never be compiled
+            (as we always wait to see all derivatives before compiling.)
+        optimize (bool, default False): whether or not to apply optimizations.
+        enabled (bool, default True): whether or not to actually enable the JIT
+            compiler.  This is a convenient way to disable a compilation statement
+            without deleting the actual `compile` invocation.
+
+    Example: Compile as function decorator.
+
+        >>> @jit.compile
+        >>> def f(x);
+        >>>     return x * 2
+        >>> x = Variable(torch.randn(1))
+        >>> out1 = f(x)  # interpreted run
+        >>> out1.sum().backward()  # required!
```

Comment:
It's not required in any way. It's fine to compute fewer derivatives, but it won't compile the code